pkg/fleet/daemon/local_api_windows.go

```diff
@@ -0,0 +1,63 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build windows
+
+package daemon
+
+import (
+	"context"
+	"github.com/Microsoft/go-winio"
+	"net"
+	"net/http"
+)
+
+const (
+	namedPipePath = "\\\\.\\pipe\\DD_INSTALLER"
+)
+
+// NewLocalAPI returns a new LocalAPI.
+func NewLocalAPI(daemon Daemon, _ string) (LocalAPI, error) {
+	listener, err := winio.ListenPipe(namedPipePath, &winio.PipeConfig{
+		// https://learn.microsoft.com/en-us/windows/win32/secauthz/security-descriptor-string-format
+		// https://learn.microsoft.com/en-us/windows/win32/secauthz/ace-strings
+		// https://learn.microsoft.com/en-us/windows/win32/secauthz/sid-strings
+		//
+		// D:dacl_flags(ace_type;ace_flags;rights;object_guid;inherit_object_guid;account_sid;(resource_attribute))
+		// 	dacl_flags:
+		//		"P": SDDL_PROTECTED
+		//	ace_type:
+		//		"A": SDDL_ACCESS_ALLOWED
+		// rights:
+		//		"GA": SDDL_GENERIC_ALL
+		// account_sid:
+		//		"WD": Everyone
+		SecurityDescriptor: "D:P(A;;GA;;;WD)",
+		MessageMode:        true,
```

Comment:
Why message mode? If we're replacing a TCP style socket, shouldn't we use byte mode?